src/core/net/tcp6.cpp

```diff
@@ -95,40 +125,69 @@ const SockAddr &Tcp::Endpoint::GetPeerAddress(void) const
 
 Error Tcp::Endpoint::Bind(const SockAddr &aSockName)
 {
-    OT_UNUSED_VARIABLE(aSockName);
+    Error error;
+    VerifyOrExit(!static_cast<const Address *>(&aSockName.mAddress)->IsUnspecified(), error = kErrorInvalidArgs);
+    VerifyOrExit(GetInstance().Get<Tcp>().CanBind(aSockName), error = kErrorInvalidState);
 
-    return kErrorNotImplemented;
+    memcpy(&mTcb.laddr, &aSockName.mAddress, sizeof(mTcb.laddr));
+    mTcb.lport = HostSwap16(aSockName.mPort);
+    error      = kErrorNone;
+
+exit:
+    return error;
 }
 
 Error Tcp::Endpoint::Connect(const SockAddr &aSockName, uint32_t aFlags)
 {
-    OT_UNUSED_VARIABLE(aSockName);
     OT_UNUSED_VARIABLE(aFlags);
 
-    return kErrorNotImplemented;
+    Error error = kErrorNone;
+
+    VerifyOrExit(mTcb.t_state == TCP6S_CLOSED, error = kErrorInvalidState);
+
+    struct sockaddr_in6 sin6p;
+    memcpy(&sin6p.sin6_addr, &aSockName.mAddress, sizeof(sin6p.sin6_addr));
+    sin6p.sin6_port = HostSwap16(aSockName.mPort);
+    error           = BsdErrorToOtError(tcp6_usr_connect(&mTcb, &sin6p));
+
+exit:
+    return error;
 }
 
 Error Tcp::Endpoint::SendByReference(otLinkedBuffer &aBuffer, uint32_t aFlags)
 {
-    OT_UNUSED_VARIABLE(aBuffer);
-    OT_UNUSED_VARIABLE(aFlags);
+    Error error;
 
-    return kErrorNotImplemented;
+    bool moreToCome = (aFlags & OT_TCP_SEND_MORE_TO_COME) != 0;
+    int  bsdError   = tcp_usr_send(&mTcb, moreToCome ? 1 : 0, &aBuffer, 0);
+    SuccessOrExit(error = BsdErrorToOtError(bsdError));
+
+exit:
+    return error;
```

Comment:
Simplify to this line?

```c++
return BsdErrorToOtError(tcp_user_send(&mTcp, (aFlags & OT_TCP_SEND_MORE_TO_COME) !=0, &aBuffer, 0);
```

Reply:
I simplified the code to a single line as you suggested.